packages/datadog-instrumentations/src/rhea.js

```diff
@@ -178,27 +177,26 @@ function patchCircularBuffer (proto, Session) {
         }
         if (CircularBuffer && !patched.has(CircularBuffer.prototype)) {
           shimmer.wrap(CircularBuffer.prototype, 'pop_if', popIf => function (fn) {
-            arguments[0] = shimmer.wrapFunction(fn, fn => AsyncResource.bind(function (entry) {
```

Comment:
@rochdev Does this look incorrect to you how I got rid of `AsyncResource.bind`. On master branch, this binded function seems to get called more than it does on this current branch, making me think I messed up on this line. I remember you saying there may be some caveats removing `.bind` calls.